lib/sycamore/sycamore/utils/http.py

```diff
@@ -0,0 +1,122 @@
+import ssl
+import random
+import mimetypes
+import http.client
+import urllib.parse
+
+from typing import Optional
+
+
+class OneShotKaClient:
+    """
+    Acts like a keepalive client, but actually closes the connection at
+    the end.  Basically, doesn't send Connection: close header as
+    urllib.request does.  This makes it more like curl.
+    """
+
+    alphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+
+    def __init__(self, url: str) -> None:
+        self.url = url
+        self.up = urllib.parse.urlparse(url)
+        self.agent = "ArynKa/1.0"
+
+    def set_agent(self, a: str) -> None:
+        self.agent = a
+
+    def get(self, headers: dict[str, str] = {}) -> bytes:
+        hdr = {
+            "User-Agent": self.agent,
+            "Host": self.up.netloc,
+            "Accept": "*/*",
+            "Accept-Encoding": "identity",
+        }
+        hdr.update(headers)
+
+        self._htconn()
+        self.conn.request("GET", self.up.path, headers=hdr)
+        return self._finish()
+
+    def post(self, *, headers: dict[str, str] = {}, form: dict[str, str] = {}) -> bytes:
```

Comment:
I think you narrowly miss [this issue](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python) by how carefully the code is written. No need for a change.

Reply:
I am not sure how this would affect multithreading.

Reply:
I was debating defaulting to None.  Perhaps it's a better policy for the future.

I don't think threading is an issue for this class, as there's no real sharing.  One instance of OneShotKaClient can be passed from thread to thread as long as only one thread uses it at a time.

Reply:
My concern is that all instances are sharing the same default dictionary and that they need to be careful to take turns changing it.